tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -169,6 +179,14 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
                 len++;
             }
 
+            var hasSpanEvents = span.SpanEvents is { Count: > 0 };
```

Comment:
nit: I noticed that you're checking `span.SpanEvents is { Count: > 0 }` multiple times. What if you create two variables `bool hasNativeSpanEvents` and `bool hasMetaSpanEvents` and then use those everywhere else?